test/functional/rpc_setban.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# Copyright (c) 2015-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the setban rpc call."""
+
+from test_framework.test_node import ErrorMatch
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    connect_nodes
+)
+
+class SetBanTests(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+        self.extra_args = [[],[]]
+
+    def run_test(self):
+        # Node 0 connects to Node 1
+        self.assert_has_permission('noban', False)
+        # Node 0 get banned by Node 1
+        self.nodes[1].setban("127.0.0.1", "add")
+        # Node 1 should not be able to reconnect
+        self.restart_node(1, [])
+        connect_nodes(self.nodes[0], 1)
+        peerinfo = self.nodes[1].getpeerinfo()
+        assert_equal(0, len(peerinfo))
+        # Node 1 should be able to reconnect if it has noban permission
+        self.restart_node(1, ['-whitelist=127.0.0.1'])
+        connect_nodes(self.nodes[0], 1)
+        self.assert_has_permission('noban', True)
+
+    def assert_has_permission(self, permission, contains):
```

Comment:
Maybe pass node in directly and split into `assert_has_permission(self, node, permission)` and `assert_has_no_permission(self, node, permission)`

Reply:
what about `assert_equal(True, self.hasPermision(...))` ?

That said, I don't think it should pass the nodes to this method. It is confusing (there is two nodes: the node that is permissioned, and the node on which we want to check permissions).

Reply:
`assert(self.hasPermision(...))` is even simpler.

You could pass both nodes. Or you could document at the top of the test file which hardcoded nodes do what.

Reply:
Sorry but I think you could just inline this - it's longer as it is with the `contains` arg.